service/history/execution/mutable_state_builder_add_continue_as_new_event_test.go

```diff
@@ -0,0 +1,344 @@
+					FirstDecisionCompletedID: 4,
+					CreatedTimeNano:          common.Ptr(int64(ts1)),
+					ExpiringTimeNano:         common.Ptr(int64(ts3)),
+					Resettable:               true,
+				}}},
+				Header: nil,
+			},
+		},
+		{
+			ID:        2,
+			Timestamp: common.Ptr(int64(ts3)),
+			EventType: common.Ptr(types.EventTypeDecisionTaskScheduled),
+			Version:   1,
+			TaskID:    -1234,
+			DecisionTaskScheduledEventAttributes: &types.DecisionTaskScheduledEventAttributes{
+				TaskList:                   &types.TaskList{Name: "helloWorldGroup"},
+				StartToCloseTimeoutSeconds: common.Ptr(int32(60)),
+			},
+		},
+	}
+
+	fetchedHistoryEvent1 := &types.HistoryEvent{
+		ID: 1, Timestamp: common.Ptr(int64(1709938156435726000)),
+		EventType: common.Ptr(types.EventTypeWorkflowExecutionStarted),
+		Version:   1,
+		TaskID:    17826364,
+		WorkflowExecutionStartedEventAttributes: &types.WorkflowExecutionStartedEventAttributes{
+			WorkflowType:                        &types.WorkflowType{Name: "helloWorldWorkflow"},
+			TaskList:                            &types.TaskList{Name: "helloWorldGroup"},
+			Input:                               []uint8{110, 117, 108, 108, 10},
+			ExecutionStartToCloseTimeoutSeconds: common.Ptr(int32(60)),
+			TaskStartToCloseTimeoutSeconds:      common.Ptr(int32(60)),
+			ContinuedExecutionRunID:             "96892ca6-975a-44b1-9726-cdb63acd8cda",
+			OriginalExecutionRunID:              "befc5b41-fb06-4a99-bec2-91c3e98b17d7",
+			FirstExecutionRunID:                 "bcdee7e4-cb21-4bbb-a8d1-43da79e3d252",
+			PrevAutoResetPoints: &types.ResetPoints{Points: []*types.ResetPointInfo{
+				{
+					BinaryChecksum:           "6df03bf5110d681667852a8456519536",
+					RunID:                    "bcdee7e4-cb21-4bbb-a8d1-43da79e3d252",
+					FirstDecisionCompletedID: 4,
+					CreatedTimeNano:          common.Ptr(int64(1709938002170829000)),
+					ExpiringTimeNano:         common.Ptr(int64(1710197212347858000)),
+					Resettable:               true,
+				},
+			}},
+			Header: &types.Header{},
+		},
+	}
+
+	tests := map[string]struct {
+		startingState *persistence.WorkflowExecutionInfo
+		// history is a substruct of current state, but because they're both
+		// pointing to each other, they're assembled at the test start
+		startingHistory []*types.HistoryEvent
+
+		// expectations
+		historyManagerAffordance func(historyManager *persistence.MockHistoryManager)
+		shardManagerAffordance   func(shardContext *shardCtx.MockContext, msb *mutableStateBuilder, domainCache cache.DomainCache)
+		domainCacheAffordance    func(domainCache *cache.MockDomainCache)
+		taskgeneratorAffordance  func(taskGenerator *MockMutableStateTaskGenerator, msb *mutableStateBuilder)
+
+		expectedReturnedState   *persistence.WorkflowExecutionInfo // this is returned
+		expectedReturnedHistory []*types.HistoryEvent
+		expectedErr             error
+	}{
+		"a continue-as-new event with no errors": {
```

Comment:
There are a few err returning if blocks in `AddContinueAsNewEvent`. Are you planning to add more cases in follow up PRs?

Reply:
Yes, that was the intent, there's quite a few non-error cases not covered as well (with child workflow data), but the data will have to be hand-created rather than snapshotted with https://github.com/hexops/valast so I was deferring it a bit because it's slow going